internal/locate/region_cache.go

```diff
@@ -330,13 +332,37 @@ func newRegion(bo *retry.Backoffer, c *RegionCache, pdRegion *pd.Region) (*Regio
 	if len(availablePeers) == 0 {
 		return nil, errors.Errorf("no available peers, region: {%v}", r.meta)
 	}
+
+	for _, p := range pdRegion.DownPeers {
+		c.storeMu.RLock()
+		store, exists := c.storeMu.stores[p.StoreId]
+		c.storeMu.RUnlock()
+		if !exists {
+			store = c.getStoreByStoreID(p.StoreId)
+		}
+		addr, err := store.initResolve(bo, c)
+		if err != nil {
+			continue
+		}
+		// Filter the peer on a tombstone store.
+		if addr == "" {
+			continue
+		}
+
+		if store.storeType == tikvrpc.TiFlash {
+			r.hasUnavailableTiFlashStore = true
```

Comment:
It seems the unavailable flag is set when loading region from PD.
If all the TiFlash stores are up when loading the region, then one of them goes down, the `hasUnavailableTiFlashStore` will be false, and the cached region might be used continuingly and may never loading again. Then the reload will also be skipped.
Correct me if I understand something wrong.

Reply:
Yes, your understanding is right. If all the TiFlash nodes are up, and there is continuous query on TiFlash, the region will not be out-dated, and if one of the TiFlash goes down, the region cache is not aware of it. But although region cache is not aware of the down node, TiDB mpp can handle this correctly because for each mpp query, TiDB will send `isAlive` rpc to all the candidate TiFlash nodes, and if fail to get response or the response is `false`, TiDB will not send task to that TiFlash node.

Reply:
> TiDB will not send task to that TiFlash node.

Even after the node is recovered, TiDB still do not send task to it?

Reply:
Once region cache can "see" the TiFlash node, TiDB will send task to it if it is back. In the case we discussed above, region cache can always "see" the TiFlash node, so TiDB will send task to it after it is recovered.